driver-extras/src/main/java/com/datastax/driver/extras/codecs/jdk8/LocalDateTimeCodec.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.driver.extras.codecs.jdk8;
+
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.exceptions.InvalidTypeException;
+
+import java.nio.ByteBuffer;
+
+import static com.datastax.driver.core.ParseUtils.isLongLiteral;
+import static com.datastax.driver.core.ParseUtils.quote;
+
+/**
+ * {@link TypeCodec} that maps
+ * {@link java.time.LocalDateTime} to CQL {@code timestamp},
+ * allowing the setting and retrieval of {@code timestamp}
+ * columns as {@link java.time.LocalDateTime} instances.
+ * <p>
+ * <strong>IMPORTANT</strong>
+ * <p>
+ * 1) The default timestamp formatter used by this codec produces CQL literals
+ * that may include milliseconds.
+ * <strong>This literal format is incompatible with Cassandra < 2.0.9.</strong>
+ * <p>
+ * 2) Even if the ISO-8601 standard accepts timestamps with nanosecond precision,
+ * Cassandra timestamps have millisecond precision; therefore, any sub-millisecond
+ * value set on a {@link java.time.LocalDateTime} will be lost when persisted to Cassandra.
+ *
+ * @see <a href="https://cassandra.apache.org/doc/cql3/CQL-2.2.html#usingtimestamps">'Working with timestamps' section of CQL specification</a>
+ */
+@IgnoreJDK6Requirement
+@SuppressWarnings("Since15")
+public class LocalDateTimeCodec extends TypeCodec<java.time.LocalDateTime> {
+
+    private static final java.time.format.DateTimeFormatter FORMATTER = java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME;
+
+    public static final LocalDateTimeCodec instance = new LocalDateTimeCodec();
+
+    private LocalDateTimeCodec() {
+        super(DataType.timestamp(), java.time.LocalDateTime.class);
+    }
+
+    @Override
+    public ByteBuffer serialize(java.time.LocalDateTime value, ProtocolVersion protocolVersion) {
+        if (value == null) {
+            return null;
+        }
+        long millis = value.atZone(java.time.ZoneOffset.UTC).toInstant().toEpochMilli();
```

Comment:
Would it make sense to make the `ZoneOffset` a final field, and expose a constructor that allows to change it? That way you could build a codec instance that matches your server's timezone.